File: Question4/Question4.cpp
#include <map>
#include <set>
#include <list>
#include <cmath>
#include <ctime>
#include <deque>
#include <queue>
#include <stack>
#include <string>
#include <bitset>
#include <cstdio>
#include <limits>
#include <vector>
#include <climits>
#include <cstring>
#include <cstdlib>
#include <fstream>
#include <numeric>
#include <sstream>
#include <iostream>
#include <algorithm>
#include <unordered_map>

using namespace std;
int totalPairs(int n, vector<int> &values)
{
    int result = 0;
    int max = 0;
    for (int left = 0; left < n-1; left++)
    {
        int end = max == left ? n : max;
        max = left + 1;
        result += 1;
        for (int right = left + 2; right < end; right++)
        {
            if (values[right - 1] > values[left])
            {
                break;
            }
            if (values[right] > values[max])
            {
                max = right;
                result += 1;
            }
        }
    }

    return result;
}

int main()
{
    int n;
    cin >> n;
    vector<int> values(n);
    for (int i = 0; i < n; ++i)
    {
        cin >> values[i];
    }
    int answer = totalPairs(n, values);
    // Do not remove below line
    cout << answer << endl;
    // Do not print anything after this line

    return 0;
}

File: Question4/Question4.py
def totalPairs(n, values):
    # Participants code will be here
    result = 0
    max = 0

    for left in range(n-1):
        end = n if max == left else max+1
        max = left+1
        result += 1
        for right in range(left+2, end):
            if values[right-1] > values[left]:
                break
            if values[right] > values[max]:
                max = right
                result += 1

    return result

if __name__ == "__main__":
    n = int(input())
    values = list(map(int, input().split()))
    answer = totalPairs(n, values)
    print(answer)